aztk/spark/utils/upload_node_scripts.py

```diff
@@ -114,5 +114,9 @@ def zip_scripts(blob_client, custom_scripts, spark_conf):
             for jar in spark_conf.jars:
                 zipf = __add_file_to_zip(zipf, jar, 'jars', binary=True)
 
+    if user_conf:
+        zipf = __add_str_to_zip(zipf, user_conf, 'user.yaml')
```

Comment:
why _add_str_to_zip? Isn't this a file?

Reply:
The user_conf is a byte string that is written to a file in node_scripts called `user.yaml`. `user.yaml` is not a file on the client. 

__add_str_to_zip() is probably not the best name for what this method does, though.